### PR TITLE
Wait for both endpoints and endpointslices in e2e tests

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/kubernetes/issues/91236

```release-note
NONE
```

/kind flake

### CODE REVIEW

File: test/e2e/network/service.go
Comment (aojea) [inline test/e2e/network/service.go:3789]: TIL from now  on we have to use List and labels always, we can not do get

**Comment by aojea** [comment]
/hold
to check the test results, 
feel free to unhold when needed

File: test/e2e/network/service.go
Comment (hasheddan) [inline test/e2e/network/service.go:3793]: Hmm looks like this is causing an issue in `pull-kubernetes-conformance-kind-ga-only-parallel`: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/93597/pull-kubernetes-conformance-kind-ga-only-parallel/1289156168177946628#1:build-log.txt%3A12459

Used here: https://github.com/kubernetes/kubernetes/blob/ee72e029070128447c673b75dbaedba978e31f48/test/e2e/network/service.go#L823

Comment (wojtek-t) [reply test/e2e/network/service.go:3793]: This is interesting - the test claims that "EndpointSlice" type is not registered there, which doesn't make much sense...

**Comment by liggitt** [comment]
This must only check endpointslices if the API exists. The GA-only conformance job disables all beta APIs and features, so it is using endpoints only

**Comment by wojtek-t** [comment]
> This must only check endpointslices if the API exists. The GA-only conformance job disables all beta APIs and features, so it is using endpoints only

Thanks Jordan - will fix in a couple minutes.

**Comment by wojtek-t** [comment]
@aojea @liggitt - PTAL

**Comment by aojea** [comment]
/lgtm
defer to Jordan to unhold it :smile:

**Review by hasheddan — APPROVED** [review]
/lgtm
/milestone v1.19

**Comment by wojtek-t** [comment]
/retest
[unrelated node-e2e failure]

File: test/e2e/network/service.go
Comment (aojea) [inline test/e2e/network/service.go:3799]: I think that there is a trailing whitespace in this line and gofmt is complaining

File: test/e2e/framework/endpointslice/ports.go
Comment (liggitt) [inline test/e2e/framework/endpointslice/ports.go:40]: nit: these lines aren't needed, can append to a nil slice (https://play.golang.org/p/pKdx39pAfLs)

no need to change this unless you need to touch the commit for other reasons

**Comment by wojtek-t** [comment]
Unrelated failure on up.
/retest

**Comment by aojea** [comment]
/test pull-kubernetes-e2e-kind 

[sig-storage] CSI mock volume CSI online volume expansion should expand volume without restarting pod if attach=on, nodeExpansion=on expand_less

**Comment by wojtek-t** [comment]
I've never seen 3 different people retesting my PR within 3 hours so far :-)

**Comment by hasheddan** [comment]
`pull-kubernetes-verify` has an issue and will continue to fail. Working on a fix 😕

**Comment by liggitt** [comment]
verify is failing on #93623, wait until that is resolved to retest

/hold

**Pull request merged**
